src/http.js

```diff
@@ -340,19 +385,51 @@ function decodeContent(html, contentType) {
  * @param {Object} options
  * @return {Promise<Object>} response, body
  */
-function customRequest(method, requestURL, options) {
+
+var getThrottledRequestForURL = (function() {
+    var i = 0;
+    return function(requestURL) {
+        console.log("REQUEST URL: " + requestURL);
+        console.log("Request Number: " + ++i);
+        //return throttledRequestsForURLs[i % 100];
+        let domain = getDomainForURL(requestURL);
+        // Customized request object exists
+        if (throttledRequestsForURLs[domain])
+            return throttledRequestsForURLs[domain];
+        // Create new customized request object if custom value exists
+        var requestIntervalForDomain = getRequestIntervalForDomain(domain);
+        if (requestIntervalForDomain) {
+            throttledRequestsForURLs[domain] = require('throttled-request')(request);
```

Comment:
This section is very similar to parts of `setupThrottledRequestForURLs` and could be made re-usable by defining a function that takes they key (e.g. 'default' or the domain), and the timeout in milliseconds.

Reply:
Done.